extensions/cornerstone-3d/src/commandsModule.js

```diff
@@ -304,11 +322,146 @@ const commandsModule = ({ servicesManager }) => {
+    },
+    async addSegmentationRepresentationToToolGroup({
+      segmentationId,
+      toolGroupId,
+      options,
+    }) {
+      const { representationType } = options;
+
+      // // Add the segmentation representation to the toolgroup
+      await cornerstoneTools.segmentation.addSegmentationRepresentations(
+        toolGroupId,
+        [
+          {
+            segmentationId,
+            type: representationType,
+          },
+        ]
+      );
+    },
+    getLabelmapVolumes: ({ segmentations }) => {
+      if (!segmentations || !segmentations.length) {
+        segmentations = SegmentationService.getSegmentations();
+      }
+
+      const labelmapVolumes = segmentations.map(segmentation => {
+        return cornerstone.cache.getVolume(segmentation.id);
+      });
+
+      return labelmapVolumes;
+    },
+    getSegmentationReport: ({ segmentations }) => {
+      if (!segmentations || !segmentations.length) {
+        segmentations = SegmentationService.getSegmentations();
+      }
+
+      let report = {};
+
+      for (const segmentation of segmentations) {
+        const { id, label, data } = segmentation;
+
+        const segReport = { id, label };
+
+        if (!data) {
+          report[id] = segReport;
+          continue;
+        }
+
+        Object.keys(data).forEach(key => {
+          if (typeof data[key] !== 'object') {
+            segReport[key] = data[key];
+          } else {
+            Object.keys(data[key]).forEach(subKey => {
+              const newKey = `${key}_${subKey}`;
+              segReport[newKey] = data[key][subKey];
+            });
+          }
+        });
+
+        const labelmapVolume = cornerstone.cache.getVolume(id);
+
+        if (!labelmapVolume) {
+          report[id] = segReport;
+          continue;
+        }
+
+        const referencedVolumeId = labelmapVolume.referencedVolumeId;
+        segReport.referencedVolumeId = referencedVolumeId;
+
+        const referencedVolume = cornerstone.cache.getVolume(
+          referencedVolumeId
+        );
+
+        if (!referencedVolume) {
+          report[id] = segReport;
+          continue;
+        }
+
+        if (!referencedVolume.imageIds || !referencedVolume.imageIds.length) {
+          report[id] = segReport;
+          continue;
+        }
+
+        const firstImageId = referencedVolume.imageIds[0];
+        const instance = OHIF.classes.MetadataProvider.get(
+          'instance',
+          firstImageId
+        );
+
+        if (!instance) {
+          report[id] = segReport;
+          continue;
+        }
+
+        report[id] = {
+          ...segReport,
+          PatientID: instance.PatientID,
+          PatientName: instance.PatientName.Alphabetic,
+          StudyInstanceUID: instance.StudyInstanceUID,
+          SeriesInstanceUID: instance.SeriesInstanceUID,
+          StudyDate: instance.StudyDate,
+        };
+      }
+
+      return report;
+    },
+    exportRTReportForAnnotations: ({ annotations }) => {
```

Comment:
Again this seems at the wrong layer, especially because its a very specific output for the RTSTRUCT in the tmtv case.